wrappers/wrapper.rb

```diff
@@ -835,17 +835,285 @@ def check_distances(vrp, unfeasible)
+                               service.activity&.duration.to_i +
+                               service.activities&.collect{ |a| a.setup_duration.to_i + a.duration.to_i }&.max.to_i
+
+            max_service_duration = service_duration if service_duration > max_service_duration
+          }
 
-      vrp
+          # NOTE: We could, in theory, add a TW to the "long" services so that they won't "block"
+          # the pause location but then we need to create alternative copies of these services
+          # for each vehicle, with different rest timewindows -- which kinda defeats the purpose.
+          #
+          # TODO: we could still simplify the other pauses of a vehicle even if some of the pauses cannot be
+          # simplified due to `max_service_duration > rest.tw` but this would complicate the post-processing
+          # e.g., we cannot shift everything later easily. At the moment, at most one pause is supported anyways.
+          next if vehicle.rests.any?{ |rest|
+            rest.timewindows.any? { |rest_tw|
+              rest_start = rest_tw.start || vehicle.timewindow&.start || 0
+              rest_end = rest_tw.end || vehicle.timewindow&.end || rest_start + vehicle.duration - rest.duration
+              max_service_duration > rest_end - rest_start
+            }
+          }
+
+          vehicle.rests.each{ |rest|
+            vehicle.duration -= rest.duration if vehicle.duration
+            vehicle.timewindow.end -= rest.duration if vehicle.timewindow&.end
+          }
+
+          vehicle[:simplified_rest_ids] = vehicle[:rest_ids].dup
+          vehicle[:rest_ids] = []
+          vehicle[:simplified_rests] = vehicle.rests.dup
+          vehicle.rests = []
+        }
+
+        vrp[:simplified_rests] = vrp.rests.select{ |r| vrp.vehicles.none?{ |v| v.rests.include?(r) } }
+        vrp.rests -= vrp[:simplified_rests]
+      when :rewind
+        # take the modifications back in case the vehicle is moved to another sub-problem
+        vrp.vehicles&.each{ |vehicle|
+          next unless vehicle[:simplified_rest_ids]&.any?
+
+          vehicle[:rest_ids].concat vehicle[:simplified_rest_ids]
+          vehicle[:simplified_rest_ids] = nil
+          vehicle.rests.concat vehicle[:simplified_rests]
+          vehicle[:simplified_rests] = nil
+
+          vehicle.rests.each{ |rest|
+            vehicle.duration += rest.duration if vehicle.duration
+            vehicle.timewindow.end += rest.duration if vehicle.timewindow&.end
+          }
+        }
+
+        if vrp[:simplified_rests]
+          vrp.rests.concat vrp[:simplified_rests]
+          vrp[:simplified_rests] = nil
+        end
+      when :patch_result
+        # correct the result with respect to simplifications
+        vrp.vehicles&.each{ |vehicle|
+          next unless vehicle[:simplified_rest_ids]&.any?
+
+          route = result[:routes].find{ |r| r[:vehicle_id] == vehicle.id }
+          no_cost = route[:activities].none?{ |a| a[:type] == 'service' }
```

Comment:
Is it okay to have shipments ? It seems we need to add this or avoid this simplification if there shipments (https://github.com/Mapotempo/optimizer-api/pull/211/files#diff-ba0217c8de9e08fe38dbb4350ea28235eaebae450a9e8881f53a9b696b702b68R929)

```suggestion
          no_cost = route[:activities].none?{ |a| ['service', 'pickup', 'delivery'].include?(a[:type]) }
```

Reply:
I agree this might be an issue even after the switch if we decide to use type: pickup for the internal services. I will do it the other way around if any none depot/pause etc.

Reply:
I modified it to be `no_cost = route[:activities].none?{ |a| %w[depot rest].exclude?(a[:type]) }`